frontend/src/routes/CriticalQuestions/fields.js

```diff
@@ -0,0 +1,60 @@
+export const fields = {
+  AGE: {
+    label: "Age",
+    name: "age",
+    initialValue: null,
+  },
+  SEX: {
+    label: "Sex",
+    name: "sex",
+    initialValue: null,
+  },
+  ETHNICITY: {
+    label: "Ethnicity",
+    name: "ethnicity",
+    initialValue: "",
+  },
+  CURRENT_LOCATION: {
+    label: "Current location",
+    name: "currentLocation",
+    initialValue: "",
+  },
+  POSTAL_CODE: {
+    label: "Postal code",
+    name: "postalCode",
+    initialValue: null,
+  },
+  COUNTRY_OF_ORIGIN: {
+    label: "Citizenship",
+    name: "countryOfOrigin",
+    initialValue: "",
+  },
+  PROFESSION: {
+    label: "Profession",
+    name: "profession",
+    initialValue: "",
+  },
+  MEDICAL_CONDITIONS: {
+    label: "Medical conditions",
+    name: "medicalConditions",
+    initialValue: [],
+  },
+  SICKNESS_START: {
+    label: "When did you first start feeling sick?",
+    name: "sicknessStart",
+    initialValue: null,
+  },
+  SICKNESS_END: {
+    label: "When did your illness resolve?",
+    name: "sicknessEnd",
+    initialValue: null,
+  },
+};
+
+export const initialFieldsState = () => {
+  const state = {};
+  for (const configs of Object.values(fields)) {
+    state[configs.name] = configs.initialValue;
+  }
+  return state;
```

Comment:
In case you wanna polish it, this could be something like:

```javascript
export const initialFieldsState = () => 
   // reducers iterate over an array and accumulate values in their first param
  Object.values(fields).reduce((state, configs) => {
    state[configs.name] = configs.initialValue;
    return state
  }, {}); // the first param is initialized as an object here
```

Reply:
yes, I know what a reducer does 😅 

I like reducer for things that should be accumulated without side-effect. 
The code you just suggested is roughly the same as what is implemented already, since it doesn't accumulate, it just alters the internal state of `state` (if not you should be returning a new state)

With your comment though I thought a more functional-way of doing this. Look at this solution
```javascript
export const initialFieldsState = () => {
  const entries = Object.values(fields).map(configs => [configs.key, configs.initialValue])
  return Object.fromEntries(entries);
};
```